File: yh_connect/src/yh_connect_sub.cpp
#include"ros/ros.h"
#include"std_msgs/Float32.h"

void msgCallback(const std_msgs::Float32::ConstPtr& msg)
{
    ROS_INFO("%f",msg->data);
}

int main(int argc,char** argv)
{
    ros::init(argc,argv,"yh_connect_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub_distance=nh.subscribe("yh_connect_float",10,msgCallback);
    //서브스크라이버선언
    //패키지(std_msgs)의메시지(String2)을이용한 서브스크라버(sub)를선언한다.
    //토픽은(yh_connect_float)이며 서브스크라이버큐(queue)사이즈를 10으로설정한다.
    //콜벡함수는 (msgCallback)이다ㅣ

    ros::spin();

    return 0;

}

File: yh_connect/scripts/yh_connect_sub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Float32

class ConnectSub:
    def __init__(self):
        self.sub=rospy.Subscriber("yh_connect_float",Float32,self.msgCallback)
     #rospy.Publisher(topic_name, msg_class, queue_size)
    #출판자혹은받는자를정의할때쓰는 형태이다
    def msgCallback(self, msg):
        rospy.loginfo(msg.data)

if __name__ =="__main__":
    rospy.init_node("yh_connect_sub")
    connect_sub=ConnectSub()
    rospy.spin()

